packages/ai-ide/src/browser/workspace-task-provider.ts

```diff
@@ -0,0 +1,137 @@
+// *****************************************************************************
+// Copyright (C) 2024 EclipseSource GmbH.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License v. 2.0 which is available at
+// http://www.eclipse.org/legal/epl-2.0.
+//
+// This Source Code may also be made available under the following Secondary
+// Licenses when the conditions for such availability set forth in the Eclipse
+// Public License v. 2.0 are satisfied: GNU General Public License, version 2
+// with the GNU Classpath Exception which is available at
+// https://www.gnu.org/software/classpath/license.html.
+//
+// SPDX-License-Identifier: EPL-2.0 OR GPL-2.0-only WITH Classpath-exception-2.0
+// *****************************************************************************
+
+import { ToolProvider, ToolRequest } from '@theia/ai-core';
+import { inject, injectable } from '@theia/core/shared/inversify';
+import { TaskService } from '@theia/task/lib/browser/task-service';
+import { TaskConfiguration } from '@theia/task/lib/common/task-protocol';
+import { ALL_TASK_TYPES } from '@theia/task/lib/browser/provided-task-configurations';
+import { TerminalService } from '@theia/terminal/lib/browser/base/terminal-service';
+import { MutableChatRequestModel } from '@theia/ai-chat';
+import { CancellationToken } from '@theia/core';
+
+const LIST_TASKS_ID = 'task';
+
+@injectable()
+export class TaskListProvider implements ToolProvider {
+
+    @inject(TaskService)
+    protected readonly taskService: TaskService;
+
+    getTool(): ToolRequest {
+        return {
+            id: `${LIST_TASKS_ID}List`,
+            name: `${LIST_TASKS_ID}List`,
+            description: 'Lists available tasks filtered by name.',
+            parameters: {
+                type: 'object',
+                properties: {
```

Comment:
I'm wondering whether the filtering by name really adds value. Usually projects have less than 30 tasks, which should be fine to always return.